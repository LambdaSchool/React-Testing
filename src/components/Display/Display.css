.component-display {
  background-color: #858694;
  color: white;
  text-align: right;
  font-size: 20px;
  font-weight: 200;
  flex: 0 0 auto;
  padding: 8px 4px;
  width: 100%;
}
