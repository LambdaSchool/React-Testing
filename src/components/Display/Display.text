Display 

prop => value must be a string; 
