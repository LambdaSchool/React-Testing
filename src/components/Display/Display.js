import React from 'react';
import PropTypes from 'prop-types';
import './Display.css';

const Display = ({ value }) => {
  return (
    <div className="component-display">
      {value}
    </div>
  );
};

Display.propTypes = {
  value: PropTypes.string.isRequired
};

export default Display;