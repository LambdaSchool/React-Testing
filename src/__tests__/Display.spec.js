import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

import Display from '../components/Display/Display';

Enzyme.configure({ adapter: new Adapter() });

describe('<Display />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Display />, div);
  });

  it('snapshot of display', () => {
    const tree = renderer.create(<Display />);
    expect(tree).toMatchSnapshot();
  });

  it('component-display exists', () => {
    const display = shallow(<Display />);
    const div = display.find('.component-display');
    expect(div.exists()).toBe(true);
  });
});
