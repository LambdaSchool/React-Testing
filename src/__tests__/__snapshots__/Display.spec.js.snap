// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Display /> snapshot of display 1`] = `
<div
  className="component-display"
>
  <div />
</div>
`;
